// DIAGNOSTICS: -UNUSED_VARIABLE

fun main() {
    val x = <!CANNOT_INFER_PARAMETER_TYPE, NEW_INFERENCE_NO_INFORMATION_FOR_PARAMETER!>run<!> <!CANNOT_INFER_PARAMETER_TYPE!>{ ::<!CANNOT_INFER_PARAMETER_TYPE, NEW_INFERENCE_NO_INFORMATION_FOR_PARAMETER!>run<!> }<!> // no error
}
